src/core/stream/representation/representation_stream.ts

```diff
@@ -377,13 +380,48 @@ export default function RepresentationStream<T>({
 
       case "parsed-segment":
         return initSegmentState.segmentData$
-          .pipe(mergeMap((initSegmentData) =>
-            pushMediaSegment({ clock$,
-                               content,
-                               initSegmentData,
-                               parsedSegment: evt.value,
-                               segment: evt.segment,
-                               segmentBuffer })));
+          .pipe(mergeMap((initSegmentData) => {
+            if (evt.value.inbandEvents !== undefined) {
+              const { inbandEvents,
+                      manifestRefreshEvent } = evt.value;
+              const manifestRefresh$ = clock$.pipe(
+                take(1),
+                mergeMap(({ position }) => {
+                  if (manifestRefreshEvent === undefined) {
+                    return EMPTY;
+                  }
+                  // redefine manifest expiration time, as typescript does not understand
+                  // that it can't be undefined when using it in the getDelay callback
+                  const { manifestExpirationTime } = manifestRefreshEvent.value;
+                  const delayComputingTime = performance.now();
+                  const getDelay = () => {
+                    const now = performance.now();
+                    const gap = (now - delayComputingTime) / 1000;
+                    return Math.max(0, manifestExpirationTime - position - gap);
```

Comment:
Why is the current position (or even here, the position at the last clock tick, without considering a `wantedTimeOffset`) important here?

Isn't just the fact that the publish `time` is now newer than our current Manifest's publish time the only thing that matters?